Makefile

```diff
@@ -86,137 +127,179 @@ clean_templates:
 	@rm -rf templates/springboot/http/target
 	@rm -f templates/**/.DS_Store
 
-.PHONY: generate/zz_filesystem_generated.go
 
-generate/zz_filesystem_generated.go: clean_templates
+# the filesysetm file itself is defined as being fresh if all the files in
+# templates are fresh as well as the certificates file.
+generate/zz_filesystem_generated.go: $(TEMPLATES) templates/certs/ca-certificates.crt
+	# Generating embedded templates filesystem
 	go generate pkg/functions/templates_embedded.go
 
-.PHONY: clean
-
-clean: clean_templates ## Remove generated artifacts such as binaries and schemas
-	rm -f $(BIN) $(BIN_WINDOWS) $(BIN_LINUX) $(BIN_DARWIN_AMD64) $(BIN_DARWIN_ARM64)
-	rm -f schema/func_yaml-schema.json
-	rm -f coverage.txt
-
-docs:
-	# Generating command reference doc
-	go run docs/generator/main.go
-
-#############
-##@ Templates
-#############
+.PHONY: templates-check
+templates-check: certs ## Check that the generated embedded filesystems are up-to-date
+	@if [[ -n $$(git diff -- schema/func_yaml-schema.json) ]]; \
```

Comment:
How does this check `generate/zz_filesystem_generated.go`?

Reply:
Yes that comment is confusing.  I removed this one and marked the templates target as .PHONY such that it always builds, the other two instances (certs and schema) were renamed to `x-modified-check` and removed the comment such that they are "hidden" from `make help`.  These are used in source control along with a `make clean` to confirm that the files are up-to-date.

Reply:
I was getting on `schema/func_yaml-schema.json`. Why it is there?

Reply:
I have removed all but the minimal makefile updates and will submit them as a separate PR for discussion